packages/fhir-router/src/batch.ts

```diff
@@ -18,7 +18,7 @@ import { IncomingHttpHeaders } from 'node:http';
 const maxUpdates = 50;
 const maxSerializableTransactionEntries = 8;
 
-const localBundleReference = /urn:uuid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/;
+const localBundleReference = /urn(:|%3A)uuid(:|%3A)[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}/;
```

Comment:
Is this actually valid?  Or is this a backwards compatibility thing?

Reply:
I believe this isn't explicitly specified between FHIR and HTTP specifications, but I gathered a few data points:
- The FHIR spec doesn't mention anything in either the [Batch HTTP](https://www.hl7.org/fhir/http.html#transaction) section or the [description of `Bundle.entry.request.url`]((http://hl7.org/fhir/R4/bundle-definitions.html#Bundle.entry.request.url)
- The [`uri` data type](http://hl7.org/fhir/R4/datatypes.html#uri) is specified to contain "A Uniform Resource Identifier Reference ([RFC 3986](http://tools.ietf.org/html/rfc3986))"
- The aforementioned RFC says about [reserved characters](https://datatracker.ietf.org/doc/html/rfc3986#section-2.2) that:
   > URI producing applications should percent-encode data octets that
   correspond to characters in the reserved set *unless these characters
   are specifically allowed by the URI scheme to represent data in that
   component*.  If a reserved character is found in a URI component and
   no delimiting role is known for that character, then it must be
   interpreted as representing the data octet corresponding to that
   character's encoding in US-ASCII. [emphasis mine]
- Specifically for [query params](https://datatracker.ietf.org/doc/html/rfc3986#section-3.4), the syntax is given as `query = *( pchar / "/" / "?" )`, which expands to the following set of characters: `ALPHA`, `DIGIT`, `-`, `.`, `_`, `~`, `!`, `$`, `&`, `'`, `(`, `)`, `*`, `+`, `,`, `;`, `=`, `:`, `@`, `/`, and `?`.  These characters do not need to be percent-encoded in the query string, but there's nothing saying they can't be